server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -1850,4 +1852,13 @@ DiscoveryNode getNode() {
             return localNode.get();
         }
     }
+
+    static {
```

Comment:
This doesn't need to be in a static block. The problem I see with statics is the exact reason they are difficult to ensure ordering of already. If something causes Node to initialize earlier, we might have an issue that can't be easily diagnosed (eg Node gets initialized before logging is setup, and a failure here then doesn't get logged correctly).

I don't feel that strongly, but I would move this to Elasticsearch.initPhase3, just before Node is constructed.

Reply:
Actually, based on the test failures you are getting here, I wonder if this needs to be in initPhase2, before we initialize the security manager. I'm not sure why it should need get classloader permissions though. Maybe @ChrisHegarty can explain?

Reply:
Yes it seems happier about being called before the security manager is installed. I moved it into `initPhase2` as you suggested.

Chris, the test failures Ryan mentions are things like this: https://gradle-enterprise.elastic.co/s/jf5zu4kc6oxlm

Reply:
The code is using a public lookup, rather than a full privileged lookup. I suspect that changing it to use a full privileged lookup should avoid triggering the security manager check, as per the overly complicated rules in [1]. e.g.

```
MethodHandles.lookup().ensureInitialized(ReferenceDocs.class);
```

[1] https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#secmgr